addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -615,9 +614,37 @@ var ImageWidget = MediaWidget.extend({
     _onSearchInput: function (ev) {
         var $input = $(ev.currentTarget);
         var $button = $input.next('.input-group-append').children();
-        var emptyValue = ($input.val() === '');
+        var inputValue = $input.val();
+        var emptyValue = (inputValue === '');
+        var $warning = this.$('.warning-url');
+        var $success = this.$('.success-url');
+        var $error = this.$('.error-url');
+        var $btn = this.$('.o_upload_media_url_button');
+        var regex = /^(?:https?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/;
+        var testURL = regex.test(inputValue);
+        var displayNone = !testURL;
+        var emptyURL = inputValue.length !== 0;
```

Comment:
- `displayNone` gives me no clue of what the variable will be used for. Why not only:
```javascript
var regex = ...;
var isURL = regex.test(inputValue);
```
then using `isURL` instead of `displayNone` ?

- `emptyURL` is useless as it is the same as `emptyValue`...